src/filters/path.rs

```diff
@@ -245,6 +245,78 @@ pub fn end() -> impl Filter<Extract = (), Error = Rejection> + Copy {
     })
 }
 
+/// Matches the end of a route with trailing slash
+/// If there is no trailing slash, then redirect to same uri with trailing slash.
+///
+/// # Example
+///
+/// ```
+/// use warp::Filter;
+///
+/// // Matches 'foo/', if 'foo' then redirect to 'foo/'
+/// let hello = warp::path::path("foo")
+///    .and(warp::path::end_with_trailing_slash_or_redirect())
+///    .map(|| "Ok");
+/// ```
+pub fn end_with_trailing_slash_or_redirect() -> impl Filter<Extract = (), Error = Rejection> + Copy
+{
+    filter_fn(move |route| {
+        if route.path().is_empty() {
+            let path = route.uri().to_string();
+            // the trailing slash rules don't apply to the website root
+            if path == "/" || path.ends_with("/") {
+                future::ok(())
+            } else {
```

Comment:
@LucianoBestia 
Hello, maybe you can reduce some nesting of `else` through `return`.

Reply:
Thank you. I also like much more this style of coding that exits the fn with "return" error soon and not nesting if-else. It looks very clean, that the happy path goes to the end of the fn code.
I squashed my commits to only one to make it easier to review. I used `git push --force` for this branch.